espnet/nets/pytorch_backend/transformer/attention.py

```diff
@@ -75,14 +75,10 @@ def forward_attention(self, value, scores, mask):
         """
         n_batch = value.size(0)
         if mask is not None:
-            mask = mask.unsqueeze(1).eq(0)  # (batch, 1, *, time2)
-            min_value = torch.finfo(scores.dtype).min
-            scores = scores.masked_fill(mask, min_value)
-            self.attn = torch.softmax(scores, dim=-1).masked_fill(
-                mask, 0.0
-            )  # (batch, head, time1, time2)
-        else:
-            self.attn = torch.softmax(scores, dim=-1)  # (batch, head, time1, time2)
+            mask = mask.unsqueeze(1)
+            mask = (~mask)*-10000
```

Comment:
Please insert a space around `*` to match the other parts

Reply:
This will change the results. So, I'm hesitating about this change.
Is it necessary?
Can you also confirm that this does not affect the result?

Also, `-10000` should be a variable and can be changed.

Reply:
I think it might not be strictly necessary, @Masao-Someki have you tested ONNX export without this change?

I do like this implementation better though, its cleaner in both the code and the ONNX computational graph in my opinion. It might affect activation values a bit but it didn't change any of the final output transcripts in our testing, but maybe more testing is needed to confirm this. Masao's codebase also uses this change.

Reply:
I like this way since it would not change the behavior depending on the program (which is important for ONNX)
However, we should at least check 
- how it would affect to the performance.
- It is better to have a unit test.

Reply:
@kenzheng99 @sw005320 
When I implemented this `-10000.0` mask, I referenced the implementation of huggingface transformers. Currently, the implementation seems to be fixed to the minimum value [link](https://github.com/huggingface/transformers/blob/4f1c9d162ee53004be4d63b76627bd0c2f5a31a9/src/transformers/models/codegen/modeling_codegen.py#L519-L525).
The output probability of onnx and torch might differ slightly (MSE < 1e-10), but I didn't observe the accuracy drop. Unit test of this parity check is implemented in the espnet_onnx repository [link](https://github.com/espnet/espnet_onnx/blob/18eb341d44ccf83c3ab35bc040b4102a73602922/tests/unit_tests/test_inference_asr.py#L58-L65).
We only need to apply the `-10000` mask for conversion, so we can use the old function while training and switch to the onnx version when conversion. I'm unsure if we can do this, but switching the mask function in `make_pad_mask` inside may be a better implementation:

```python
def make_pad_mask(...):
  if onnx_converion:
    return make_pad_mask_onnx(...)
  else:
    # old implementation
```